### PR TITLE
Initial design for new weblistener

### PR DESCRIPTION
**Pull request by ckousik**
unfinished...just the initial idea...

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "description":  "Client for interfacing taskcluster components",
   "license":      "MPL-2.0",
   "scripts": {
-    "test":       "./test/runtests.sh"
+    "test":       "./test/runtests.sh",
+    "compile":      "babel lib --out-dir build"
```
Comment (jonasfj) [inline package.json]: @imbstack, what's the trick to setup `babel-compile` with `babel-taskcluster-preset` ?

File: lib/wslistener.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+/*
+  options : {
+    baseUrl:  ....
+  }
+*/
+
+class WSListener extends EventEmitter {
+
+  constructor (options) {
+
```
Comment (jonasfj) [inline lib/wslistener.js]: No need for empty line at start or end of function...

But I like that you split the function into segments with some empty lines :)

File: lib/wslistener.js [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    if (this.socket && this.socket.readyState === READYSTATE.OPEN) {
+      return this._send('bind', binding);
+    }
+    return Promise.resolve(undefined);
+  }
+
+  //Experimental -- unfinished
+  unbind (binding) {
+    _.remove(this._bindings, b => b === binding);
```
Comment (jonasfj) [inline lib/wslistener.js]: object equality probably isn't the best option here...

Comment (jonasfj) [inline lib/wslistener.js]: I think it's reference based equality you're doing...

File: lib/wslistener.js [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    this._bindings = [];
+    this._pendingPromises = {};
+
+    this.onMessage = this.onMessage.bind(this);
+    this.onError = this.onError.bind(this);
+    this.onClose = this.onClose.bind(this);
+
+  }
+
+  connect () {
```
Comment (jonasfj) [inline lib/wslistener.js]: probably this can be made async and a lot prettier :)

**Comment by jonasfj** [comment]
looks like the right direction...

**Comment by djmitche** [comment]
WebListener is now in tc-client-web, AIUI, and uses the new tc-events.

**Pull request closed without merging**